spec/src/main/asciidoc/repository.asciidoc

```diff
@@ -112,6 +112,246 @@ Subsequently, an implementation of Jakarta Data, specifically tailored to the ch
 
 Jakarta Data empowers developers to shape their data access strategies by defining entity classes and repositories, with implementations seamlessly adapting to the chosen data store. This flexibility and Jakarta Data's persistence-agnostic approach promote robust data management within Java applications.
 
+=== Repository interfaces
+
+A Jakarta Data repository is a Java interface annotated `@Repository`.
+A repository interface may declare:
+
+- _abstract_ (non-`default`) methods, and
+- _concrete_ (`default`) methods.
+
+A concrete method may call other methods of the repository, including abstract methods.
+
+Every abstract method of the interface is usually either:
+
+- an entity instance _lifecycle method_,
+- an _annotated query method_,
+- an _automatic query method_, or
+- a _resource accessor method_.
+
+A repository may declare lifecycle methods for a single entity type, or for multiple related entity types.
+Similarly, a repository might have query methods which return different entity types.
+
+A repository interface may inherit methods from a superinterface.
+A Jakarta Data implementation must treat inherited abstract methods as if they were directly declared by the repository interface.
+For example, a repository interface may inherit the `CrudRepository` interface defined by this specification.
```

Comment:
> we need to specify how the user identifies the entity types to the Jakarta Data provider

My preference would be that: (at least for ):

- there simply is no primary entity for repositories which don't inherit `BasicRepository`, and
- instead, there's a way to infer the entity type for each abstract method of the repository. 

I _believe_ that's doable. The return type of a query method should be enough to infer the queried entity  type.

But we need to talk that through because I might be missing something.

Otherwise your suggested change is fine. 👍

Reply:
> The return type of a query method should be enough to infer the queried entity type.

That should work for `find` queries, and for queries with `@Query` that return some other type, the entity type wouldn't be needed because the query language is hard-coded.  The only places I can think of where the entity type is needed but isn't determinable from the method signature are the `countBy` and `existsBy` methods, and  I agree it doesn't make sense to be able to define such methods when there isn't a primary entity type.  I'll see if I can update my prior code suggestion with that.

Reply:
That worked. My above suggestion is updated now.

Reply:
OK cool, I will take a look later.